homeassistant/components/backup/manager.py

```diff
@@ -347,9 +355,15 @@ async def async_setup(self) -> None:
             self.known_backups.load(stored["backups"])
 
         await self._reader_writer.async_validate_config(config=self.config)
-        await self._reader_writer.async_resume_restore_progress_after_restart(
-            on_progress=self.async_on_backup_event
-        )
+
+        async def set_manager_idle_after_start(hass: HomeAssistant) -> None:
+            """Set manager to idle after start."""
+            self.async_on_backup_event(IdleEvent())
+            await self._reader_writer.async_resume_restore_progress_after_restart(
+                on_progress=self.async_on_backup_event
+            )
```

Comment:
The delayed resume of the `RESTORE_BACKUP` state may impact frontend. Maybe we should transition directly to that state and skip the `DISABLED` state if core is restarting because of an ongoing restore.

Reply:
Yes, if nothing is blocking the backup restore after restart it's ok to resume directly and transition to RESTORE_BACKUP.